python/lsst/obs/base/ingest.py

```diff
@@ -1198,12 +1209,26 @@ def ingestFiles(
                     str(list(inserted_or_updated.keys())),
                 )
 
+            # Determine the instrument so we can work out the dataset type.
+            instrument = exposure.files[0].instrument
+            assert (
+                instrument is not None
+            ), "file should have been removed from this list by prep if instrument could not be found"
+
+            if raw_definition := getattr(instrument, "raw_definition", None):
+                datasetTypeName = raw_definition[0]
+                if not (datasetType := datasetTypes.get(datasetTypeName)):
+                    datasetType = DatasetType(
+                        *instrument.raw_definition, universe=self.butler.registry.dimensions
+                    )
+            else:
+                datasetType = self.datasetType
+            if datasetType not in datasetTypes:
```

Comment:
```suggestion
            if datasetType.name not in datasetTypes:
```

Reply:
Thanks. Embarrassingly that bug completely ruins the caching...